### PR TITLE
Maintenance: Use #defines to improve code readability

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Use `#define` to improve code readability. 
- for `UIUserInterfaceIdiom` checks
- for `UIInterfaceOrientation` checks

## Summary for release notes

Maintenance: Use #defines for device type and device orientation checks

### CODE REVIEW

File: XBMC Remote/AppDelegate.h [hunk 1]
```diff
@@ -31,6 +31,8 @@
 
 #define IS_AT_LEAST_IPHONE_X_HEIGHT (CGRectGetHeight(UIScreen.mainScreen.fixedCoordinateSpace.bounds) >= 812)
 #define IS_AT_LEAST_IPAD_1K_WIDTH (CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds) >= 1024)
+#define IS_IPHONE ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone)
+#define IS_IPAD ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad)
```
Comment (kambala-decapitator) [inline XBMC Remote/AppDelegate.h:35]:
```suggestion
#define IS_IPHONE (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPhone)
#define IS_IPAD (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPad)
```

File: XBMC Remote/AppDelegate.h [hunk 2]
```diff
@@ -33,6 +33,8 @@
 #define IS_AT_LEAST_IPAD_1K_WIDTH (CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds) >= 1024)
 #define IS_IPHONE ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone)
 #define IS_IPAD ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad)
+#define IS_PORTRAIT (UIInterfaceOrientationIsPortrait([[UIApplication sharedApplication] statusBarOrientation]))
+#define IS_LANDSCAPE (UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation]))
```
Comment (kambala-decapitator) [inline XBMC Remote/AppDelegate.h:37]:
```suggestion
#define IS_PORTRAIT UIInterfaceOrientationIsPortrait(UIApplication.sharedApplication.statusBarOrientation)
#define IS_LANDSCAPE UIInterfaceOrientationIsLandscape(UIApplication.sharedApplication.statusBarOrientation)
```

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
looks good apart from minor style changes.

Suggestions for the future:

1. move convenience macros out of `AppDelegate.h` to a separate header and include it in a PCH
2. try using proper portrait/landscape checks instead of the status bar "hack"

File: XBMC Remote/AppDelegate.h (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/AppDelegate.h:35]: :) I was thinking about this and even started working on changing it at many other places. Finally, this was too much of a change and I reverted everything. Will apply the changes for this case though.

File: XBMC Remote/AppDelegate.h (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/AppDelegate.h:37]: Same here.

**Comment by wutschel** [comment]
> try using proper portrait/landscape checks instead of the status bar "hack"

Here we go. If ok, I will squash and push force.

**Comment by wutschel** [comment]
> move convenience macros out of `AppDelegate.h` to a separate header and include it in a PCH

There is a PCH defined already, so I moved some defines into a new header and imported it from there.

File: XBMC Remote/AppDelegate.h [hunk 3]
```diff
@@ -33,8 +33,8 @@
 #define IS_AT_LEAST_IPAD_1K_WIDTH (CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds) >= 1024)
 #define IS_IPHONE (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPhone)
 #define IS_IPAD (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPad)
-#define IS_PORTRAIT (UIInterfaceOrientationIsPortrait([[UIApplication sharedApplication] statusBarOrientation]))
-#define IS_LANDSCAPE (UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation]))
+#define IS_PORTRAIT (UIInterfaceOrientationIsPortrait(UIApplication.sharedApplication.statusBarOrientation))
+#define IS_LANDSCAPE (UIInterfaceOrientationIsLandscape(UIApplication.sharedApplication.statusBarOrientation))
```
Comment (kambala-decapitator) [inline XBMC Remote/AppDelegate.h]:
```suggestion
#define IS_PORTRAIT UIInterfaceOrientationIsPortrait(UIApplication.sharedApplication.statusBarOrientation)
#define IS_LANDSCAPE UIInterfaceOrientationIsLandscape(UIApplication.sharedApplication.statusBarOrientation)
```

File: XBMC Remote/AppDelegate.h [hunk 4]
```diff
@@ -33,8 +33,8 @@
 #define IS_AT_LEAST_IPAD_1K_WIDTH (CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds) >= 1024)
 #define IS_IPHONE (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPhone)
 #define IS_IPAD (UIDevice.currentDevice.userInterfaceIdiom == UIUserInterfaceIdiomPad)
-#define IS_PORTRAIT (UIInterfaceOrientationIsPortrait(UIApplication.sharedApplication.statusBarOrientation))
-#define IS_LANDSCAPE (UIInterfaceOrientationIsLandscape(UIApplication.sharedApplication.statusBarOrientation))
+#define IS_PORTRAIT (UIDeviceOrientationIsPortrait(UIDevice.currentDevice.orientation))
+#define IS_LANDSCAPE (UIDeviceOrientationIsLandscape(UIDevice.currentDevice.orientation))
```
Comment (kambala-decapitator) [inline XBMC Remote/AppDelegate.h]: checking device orientation for this is a bad solution. Device orientation is needed only to determine device orientation in 3D space, i.e. physical world.

By proper approach I meant something like comparing view's width to its height. Ideally, we should use the regular and compact traits. But that's definitely not for this PR.

File: XBMC Remote/AppDelegate.h [hunk 5]
```diff
@@ -23,11 +23,12 @@
  *  System Versioning Preprocessor Macros
  */
 
-#define SYSTEM_VERSION_EQUAL_TO(v)                  ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedSame)
-#define SYSTEM_VERSION_GREATER_THAN(v)              ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedDescending)
-#define SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(v)  ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] != NSOrderedAscending)
-#define SYSTEM_VERSION_LESS_THAN(v)                 ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] == NSOrderedAscending)
-#define SYSTEM_VERSION_LESS_THAN_OR_EQUAL_TO(v)     ([[[UIDevice currentDevice] systemVersion] compare:v options:NSNumericSearch] != NSOrderedDescending)
+#define SYSTEM_VERSION_GET(v)                       ([UIDevice.currentDevice.systemVersion compare:v options:NSNumericSearch])
```
Comment (kambala-decapitator) [inline XBMC Remote/AppDelegate.h]: forget about `UIDevice.currentDevice.systemVersion`, use `NSProcessInfo`: either `operatingSystemVersion` property or `isOperatingSystemAtLeast...` method. These macros should also be removed.

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: XBMC Remote/AppDelegate.h (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/AppDelegate.h]: Ok, will apply this change.

File: XBMC Remote/AppDelegate.h (hunk 4 above)
Comment (wutschel) [reply XBMC Remote/AppDelegate.h]: Ok, will then go back to `UIInterfaceOrientationIsPortrait(UIApplication.sharedApplication.statusBarOrientation`.

File: XBMC Remote/AppDelegate.h (hunk 5 above)
Comment (wutschel) [reply XBMC Remote/AppDelegate.h]: There is only call of `if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"11.0"))` left. At all other places simply `if (@available(iOS 11.0, *))` is used. Wouldn´t this work as well?

Comment (kambala-decapitator) [reply XBMC Remote/AppDelegate.h]: `@available` does work, but I prefer to use it only for API checks, for simple runtime OS check I always use `NSProcessInfo`. Not insisting on changing :)

**Pull request merged**